modules/caddyhttp/encode/zstd/zstd.go

```diff
@@ -51,7 +85,7 @@ func (z Zstd) NewEncoder() encode.Encoder {
 	// The default of 8MB for the window is
 	// too large for many clients, so we limit
 	// it to 128K to lighten their load.
-	writer, _ := zstd.NewWriter(nil, zstd.WithWindowSize(128<<10), zstd.WithEncoderConcurrency(1), zstd.WithZeroFrames(true))
+	writer, _ := zstd.NewWriter(nil, zstd.WithWindowSize(128<<10), zstd.WithEncoderConcurrency(1), zstd.WithZeroFrames(true), zstd.WithEncoderLevel(z.Level))
```

Comment:
This line's kinda long, let's adjust the syntax

```suggestion
	writer, _ := zstd.NewWriter(
		nil,
		zstd.WithWindowSize(128<<10), 
		zstd.WithEncoderConcurrency(1), 
		zstd.WithZeroFrames(true), 
		zstd.WithEncoderLevel(z.Level),
	)
```